src/Commands/core/NotifyCommands.php

```diff
@@ -88,9 +88,10 @@ public static function shutdownSendText($msg, CommandData $commandData)
             }
         }
 
-        if (!drush_shell_exec($cmd, $msg)) {
-            throw new \Exception($error_message . ' ' . dt('Or you may specify an alternate command to run by setting notify:cmd in a drush.yml file'));
-        }
+        // Keep backward compat and prepare a string here.
+        $cmd = sprintf($cmd, Escape::forSite(Drush::aliasManager()->getSelf(), $msg));
+        $process = Drush::process($cmd);
```

Comment:
Why not omit Escape::forSite and just:
`$process = Drush::process($cmd, [$msg])`?

Reply:
the 2nd argument to process() is cwd, not a replacement array. And if there were two items in the $cmd array, it would quote the first long item which is invalid I think.

Reply:
My mistake. I meant:
```
$process = Drush::process([$cmd, $msg]);
```
That will escape the commands for you. Hm, that uses Symfony escaping instead of our improved escaping. Maybe we should have Drush::process use the Escape utilities to escape all of the parameters for the local machine. `Escape::shellArg()` is the API to use to just escape for the local system.